sdks/go/pkg/beam/runners/prism/internal/worker/worker.go

```diff
@@ -710,3 +660,130 @@ func (wk *W) MonitoringMetadata(ctx context.Context, unknownIDs []string) *fnpb.
 		},
 	}).GetMonitoringInfos()
 }
+
+type MultiplexW struct {
+	fnpb.UnimplementedBeamFnControlServer
+	fnpb.UnimplementedBeamFnDataServer
+	fnpb.UnimplementedBeamFnStateServer
+	fnpb.UnimplementedBeamFnLoggingServer
+	fnpb.UnimplementedProvisionServiceServer
+
+	mu       sync.Mutex
+	endpoint string
+	logger   *slog.Logger
+	pool     map[string]*W
+}
+
+// New instantiates a new MultiplexW for multiplexing FnAPI requests to a W.
+func New(lis net.Listener, g *grpc.Server, logger *slog.Logger) *MultiplexW {
+	_, p, _ := net.SplitHostPort(lis.Addr().String())
+	mw := &MultiplexW{
```

Comment:
Note that in *this* context, `mw` is a good name, it's used many times in a short context, meaning a shorter name is more readable. The scope is just barely a dozen lines long.